projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,234 @@
+#include "fan_control_solar.h"
+
+#include <stdbool.h>
+
+#include "data_store.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fault_handler.h"
+#include "fault_monitor.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "ms_test_helpers.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Any defines here
+
+// Declare static test variables
+static GpioAddress s_overtemp_addr;
+static GpioAddress s_fan_fail_addr;
+static GpioAddress s_full_speed_addr;
+static uint16_t s_full_speed_temp_threshold;
+static SolarMpptCount s_mppt_count;
+
+static GpioState s_gpio_state;
+static Event s_e;
+
+static GpioAddress s_dummy;
+
+static void TEST_MOCK(prv_fanfail_callback)(const GpioAddress *address, void *context) {
+  LOG_WARN("fan_control detected fanfail, raising fault event");
+  fault_handler_raise_fault(EE_SOLAR_FAULT_FAN_FAIL, 0);
+}
+
+void setup_test(void) {
+  event_queue_init();
+  gpio_init();
+  gpio_it_init();
+  data_store_init();
+
+  // Test Board A with 6 MPPTs uses virtual port A
+  GpioAddress s_overtemp_addr = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress s_fan_fail_addr = { .port = GPIO_PORT_A, .pin = 1 };
+  GpioAddress s_full_speed_addr = { .port = GPIO_PORT_A, .pin = 2 };
+
+  // arbitrarily selected value for testing
+  uint16_t s_full_speed_temp_threshold = 3;
+
+  SolarMpptCount s_mppt_count = SOLAR_BOARD_6_MPPTS;
+}
+
+// Used to reinitialize static testing variables Event s_e & GpioState s_gpio_state
+void teardown_test(void) {
+  static GpioState s_gpio_state;
+  static Event s_e;
+}
+
+// Test that "overtemperature" & "fan fail" faults are raised properly
+void test_fan_fail_fault_handling(void) {
+  FanControlSettingsSolar settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  // Test that there are no faults initially
+  MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();
+
+  // Test that fan fail fault is raised correctly
+  // The MAX6643 will call the prv_callback function when fan fail is detected
+  // Parameters have no effect on function
+  prv_fanfail_callback(&s_dummy);
+  event_process(&s_e);
+  MS_TEST_HELPER_ASSERT_EVENT(s_e, EE_SOLAR_FAULT_FAN_FAIL, 0);
```

Comment:
You don't need to `event_process` to get the next event before passing it to `MS_TEST_HELPER_ASSERT_EVENT` - if you take a look at the definition inside ms_test_helpers.h, `MS_TEST_HELPER_ASSERT_EVENT` does `event_process` on the event you pass itself.